### PR TITLE
[JENKINS-51913] Prevent `archiveArtifacts` from displaying a stack trace when no artifacts are found

### PR DESCRIPTION
**Pull request by gregswift**
Re-implementation of #4913 
See [JENKINS-51913](https://issues.jenkins-ci.org/browse/JENKINS-51913).

### Testing
Added a test to validate that there isn't a stack trace, and that the error is present.  

### Proposed changelog entries

Do not print a stack trace when archiveArtifact does not find a file.

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [X] (If applicable) Jira issue is well described
- [X] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [X] Appropriate autotests or explanation to why this change has no tests
- [x] (n/a) For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers
@basil @jackgene 

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### Misc

Shout out to @adrianjgeorge for helping me out :D

### CODE REVIEW

**Comment by gregswift** [comment]
thanks @basil for the assist on the errors in the test.

**Review by basil — APPROVED** [review]
Thanks for the PR!

**Review by timja — APPROVED** [review]

**Comment by basil** [comment]
This PR is now ready for merge. We will merge it after approximately 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process. Thanks!

File: test/src/test/java/hudson/tasks/ArtifactArchiverTest.java
Comment (jglick) [inline test/src/test/java/hudson/tasks/ArtifactArchiverTest.java:145]: Or more directly could do something along the lines of

```java
assertThat("No stacktrace shown", build.getLog(), not(containsString("\tat ")));
```

**Review by jglick — COMMENTED** [review]
Thanks! I saw the ping a couple weeks ago but just did not have time to dig in.

File: test/src/test/java/hudson/tasks/ArtifactArchiverTest.java
Comment (basil) [reply test/src/test/java/hudson/tasks/ArtifactArchiverTest.java:145]: But then that would be inconsistent with other code in this file.

**Pull request merged**
